src/main/java/com/mraof/minestuck/entity/AnimatedCreatureEntity.java

```diff
@@ -76,74 +36,14 @@ public void tick()
 		if(animationTicks > 0)
 		{
 			animationTicks--;
-			if(animationTicks == 0 && getCurrentAction() == Actions.ATTACK)
-			{
-				animationTicks = attackRecovery;
-				this.entityData.set(CURRENT_ACTION, Actions.ATTACK_RECOVERY.ordinal());
-				performAttack();
-			} else if(animationTicks == 0)
-			{
-				this.entityData.set(CURRENT_ACTION, Actions.NONE.ordinal());
-			}
+			if(animationTicks == 0)
+				this.endTimedAction(this.getCurrentAction());
 		}
 	}
 	
-	private void performAttack()
+	protected void endTimedAction(Actions action)
 	{
```

Comment:
title is still open to different interpretations from my view, can you add javadoc?